instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb

```diff
@@ -0,0 +1,141 @@
+            attributes = {
+              'component' => 'template',
+              'operation' => 'endpoint_render'
+            }
+            tracer.in_span(name, attributes: attributes, start_timestamp: start, kind: :server) do |span|
+              handle_error(span, payload[:exception_object]) if payload[:exception_object]
+            end
+          end
+
+          # Handles the endpoint_run_filters.grape events
+          def endpoint_run_filters(_name, start, finish, _id, payload)
+            filters = payload[:filters]
+            type = payload[:type]
+
+            # Prevent submitting empty filters
+            return if (!filters || filters.empty?) || !type || (finish - start).zero?
+
+            name = span_name(payload[:endpoint])
+            attributes = {
+              'component' => 'web',
+              'operation' => 'endpoint_run_filters',
+              'grape.filter.type' => type.to_s
+            }
+            tracer.in_span(name, attributes: attributes, start_timestamp: start, kind: :server) do |span|
+              handle_error(span, payload[:exception_object]) if payload[:exception_object]
+            end
+          end
+
+          # Handles the format_response.grape event
+          def format_response(_name, start, _finish, _id, payload)
+            endpoint = payload[:env]['api.endpoint']
+            name = span_name(endpoint)
+            attributes = {
+              'component' => 'template',
+              'operation' => 'format_response',
+              'grape.formatter.type' => formatter_type(payload[:formatter])
+            }
+            tracer.in_span(name, attributes: attributes, start_timestamp: start, kind: :server) do |span|
+              handle_error(span, payload[:exception_object]) if payload[:exception_object]
+            end
+          end
+
+          private
+
+          def tracer
+            Grape::Instrumentation.instance.tracer
+          end
+
+          def span_name(endpoint)
+            "#{api_instance(endpoint)} #{request_method(endpoint)} #{path(endpoint)}"
+          end
+
+          def run_attributes(payload)
+            endpoint = payload[:endpoint]
+            path = path(endpoint)
+            {
+              'component' => 'web',
+              'operation' => 'endpoint_run',
+              'grape.route.endpoint' => api_instance(endpoint),
+              'grape.route.path' => path,
+              'grape.route.method' => endpoint.options[:method].first,
+              OpenTelemetry::SemanticConventions::Trace::HTTP_METHOD => request_method(endpoint),
+              OpenTelemetry::SemanticConventions::Trace::HTTP_ROUTE => path
+            }
+          end
+
+          def handle_error(span, exception)
+            span.record_exception(exception)
+            span.status = OpenTelemetry::Trace::Status.error("Unhandled exception of type: #{exception.class}")
+            return unless exception.respond_to?('status') && exception.status
```

Comment:
Yes, this is something I added because indeed http.status_code was not in the DDog instrumentation.

The reason why I added it was to have information about errors that are not exceptions without having to set the [`error_statuses` config provided by DDog](https://docs.datadoghq.com/tracing/trace_collection/dd_libraries/ruby/#grape). I thought that using the HTTP status code as an attribute would help with filtering/querying errors without needing to resort to custom configs and parsing logic like DDog does.
I'm totally open to removing it as long as it doesn't mean we need to add the error statuses config and the parser 😱 But I'd like to ask the maintainers what they think.

To summarize, there are 4 options:
- Not add the `error_statuses` config and add HTTP status code to also be able to filter errors that don't raise an exception.
- Not add `error_statuses` and skip the HTTP status code too, even if that means we can't filter errors that don't raise an exception in the Grape spans (I'm ok with this too, but I'd like to document it).
- Rely on the `error_statuses` config as it's done in DDog and not add the HTTP status code (I'd rather avoid this, but maybe the maintainers want to stay close to the DDog implementation)
- Add the `error_statuses` config AND the HTTP status code (which IMO this would be too much).

If you agree, I can share this with the maintainers and we can decide based on their feedback, wdyt?

Reply:
>The reason why I added it was to have information about errors that are not exceptions without having to set the [error_statuses config provided by DDog](https://docs.datadoghq.com/tracing/trace_collection/dd_libraries/ruby/#grape). I thought that using the HTTP status code as an attribute would help with filtering/querying errors without needing to resort to custom configs and parsing logic like DDog does.
I'm totally open to removing it as long as it doesn't mean we need to add the error statuses config and the parser 😱

so the exception raised here would be an error from the perspective of the endpoint implementation. which can indeed be a false positive like you said. The error from a response perspective would be more well defined once all the middleware finish running. I saw that the rack middleware sets the error based on status codes which would allow a user to get that information using that middleware.

IMO this is not something we (grape instrumentation) should be concerned about as it would be handled by a higher level instrumentation like rack or rails or sinatra.

>But I'd like to ask the maintainers what they think.

that sounds good though 👍